steampy/client.py

```diff
@@ -241,7 +241,68 @@ def make_offer(self, items_from_me: List[Asset], items_from_them: List[Asset], p
         if response.get('needs_mobile_confirmation'):
             return self._confirm_transaction(response['tradeofferid'])
         return response
+        
+#takes a trade url and returns the maximum escrow duration between u and ur partner
+    @login_required
+    def get_escrow_duration(self, trade_offer_url: str) -> int:
+
+        start = trade_offer_url.index('steamcommunity.com') + len('steamcommunity.com')
+        end_trade_url=trade_offer_url[start:]
+
+        headers = {'Referer': self.COMMUNITY_URL + end_trade_url,
+                   'Origin': self.COMMUNITY_URL}
+        response = self._session.get(trade_offer_url, headers=headers).text
+        my_escrow_duration=int(text_between(response,"var g_daysMyEscrow = ",";"))
+        their_escrow_duration=int(text_between(response,"var g_daysTheirEscrow = ",";"))
+        
+        return max(my_escrow_duration,their_escrow_duration)
+
 
+    #same as make_offer but takes a trade url of your partner and allows u to send offer as non friends
+    @login_required
+    def make_offer_url(self, items_from_me: List[Asset], items_from_them: List[Asset], trade_offer_url: str, partner_steam_id: str,
```

Comment:
You don't need patner_steam_id because in trade_offer_url you have account id (in tests 314218906) and you can use
steam_id = utils.account_id_to_steam_id(account_id) and dont pass it as parameter

Rename it to make_offer_with_trade_url